### PR TITLE
Fix bugs when run YamlSerialization in multi-threads

### PR DESCRIPTION
**Pull request by qinezh**
As title, @chenkennt @vwxyzh @superyyrrzz @hellosnow @ansyral

### CODE REVIEW

**Comment by vwxyzh** [comment]
👍

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.YamlSerialization/NodeDeserializers/EmitArrayNodeDeserializer.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public class EmitArrayNodeDeserializer : INodeDeserializer
 #else
             typeof(EmitArrayNodeDeserializer).GetMethod(nameof(DeserializeHelper));
 #endif
-        private static readonly Dictionary<Type, Func<EventReader, Type, Func<EventReader, Type, object>, object>> _funcCache =
+        private readonly Dictionary<Type, Func<EventReader, Type, Func<EventReader, Type, object>, object>> _funcCache =
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.YamlSerialization/NodeDeserializers/EmitArrayNodeDeserializer.cs]: is concurrentDictionary better?

**Comment by qinezh** [comment]
Updated @chenkennt

**Comment by chenkennt** [comment]
>             if (!_funcCache.TryGetValue(expectedType, out func))

then should use TryAdd?

---
Refers to: src/Microsoft.DocAsCode.YamlSerialization/NodeDeserializers/EmitArrayNodeDeserializer.cs:36 in a809001. [](commit_id = a80900123f7e4aa55926e3b6e377fb18f999eba7, deletion_comment = False)

**Comment by superyyrrzz** [comment]
:+1:

**Pull request merged**